packages/eas-cli/src/project/publish.ts

```diff
@@ -193,35 +197,60 @@ export async function buildBundlesAsync({
     throw new Error('Could not locate package.json');
   }
 
-  const platformArgs =
-    platformFlag === 'all'
-      ? ['--platform', 'ios', '--platform', 'android']
-      : ['--platform', platformFlag];
-
-  if (shouldUseVersionedExpoCLI(projectDir, exp)) {
-    await expoCommandAsync(projectDir, [
+  // Legacy global Expo CLI
+  if (!shouldUseVersionedExpoCLI(projectDir, exp)) {
+    return await expoCommandAsync(projectDir, [
       'export',
       '--output-dir',
       inputDir,
+      '--experimental-bundle',
+      '--non-interactive',
       '--dump-sourcemap',
       '--dump-assetmap',
-      ...platformArgs,
+      ...platformFlag,
```

Comment:
Don't we need
```suggestion
      '--platform',
      ...platformFlag,
```
here?

Reply:
Yep, you are right. I just made it a bit more verbose, where all `platformFlag` instances are now `--platform=${platformFlag}`. So it's better spottable where we need to set it like this.

Reply:
I can change it back, but there were 2 instances where we needed to do this.